### ISSUE TITLE
ideas for packaging as rpm/deb

### ISSUE THREAD

**Issue by cjw296** [conda/conda#1266]
Okay, so this forms my official feature request for a .deb or .rpm to install conda system-wide so that sysadmins have an easier way to provide access to conda.

@tswicegood requested I dump this here folllowing our chat at PyCon...

So, the ideas we talked about:
- have the rpm/deb create a conda user
- have conda run as that user

This enables a shared package cache that only the conda user can write to but all conda users can read from, this means that we get the win of only pulling packages over the network once for each machine.

The only rough edge I can think of is whether conda running as the conda user could write to the current users home directory, if that's where they want their conda environment created.

**Comment by tswicegood** [comment]
Assigning this one to myself so I can keep track of it.

----

@bkreider / @lilmatt: any thoughts on how to deal with a `conda` user that should be able to write to a user's home directory on debian or redhat?

**Comment by tswicegood** [comment]
To add a little more context, I believe this will be easy once something like #1145 is done.  At that point, Conda becomes something that works on an installation of Conda rather than is part of a Conda Installation/System.

As it stands, it's definitely harder to do now because Conda assumes it's got its own version of Python that it completely controls.  It's not impossible with the current code base, but the current implementation does mean there are more moving parts to get right if we offer this.

**Comment by bkreider** [comment]
I don't think it is feasible to have a shared package cache that any user can update.  It's a security nightmare, since you can specify arbitrary channels and any user can force it to pull trojaned version of a package from malicious repo.  Signed and fully namespaced packages would probably help with that.  

> any thoughts on how to deal with a conda user that should be able to write to a user's home directory on debian or redhat?

I think when a user runs `conda` it should run as their user and do what it needs to do.  The package cache sync/pull is a separate issue.  The create or update conda environments step could still be run as the current user.  

I guess a caching web proxy would be similar to the functionality.

**Comment by tswicegood** [comment]
Security definitely could be an issue.

@cjw296 one of the main things you were talking about was internal package X getting pulled down by every single user because they all need it.  Would having a way to prime the package cache be enough?  Thinking of it in terms of Git, you fetch commits, then merge them.  The same could be done for Conda packages, except it's fetch then install.
